src/bot/notifier.test.ts

```diff
@@ -0,0 +1,69 @@
+import { jest } from "@jest/globals";
+
+describe("notifier", () => {
+  describe("OTP validation regex", () => {
+    it("should validate correct OTP formats", () => {
+      const otpRegex = /^\d{4,8}$/;
+
+      // Valid OTP codes
+      expect("1234").toMatch(otpRegex);
+      expect("123456").toMatch(otpRegex);
+      expect("12345678").toMatch(otpRegex);
+
+      // Invalid OTP codes
+      expect("123").not.toMatch(otpRegex); // too short
+      expect("123456789").not.toMatch(otpRegex); // too long
+      expect("12a456").not.toMatch(otpRegex); // contains letters
+      expect("12-456").not.toMatch(otpRegex); // contains dash
+      expect("").not.toMatch(otpRegex); // empty
+    });
+  });
+
+  describe("requestOtpCode function", () => {
+    it("should exist and be a function", async () => {
+      // Mock the dependencies to avoid module loading issues
+      jest.doMock("../config.js", () => ({
+        config: {
+          options: {
+            notifications: {
+              telegram: {
+                apiKey: "test-api-key",
+                chatId: "test-chat-id",
+                enableOtp: true,
+                otpTimeoutSeconds: 300,
+              },
+            },
+            scraping: {
+              hiddenDeprecations: [],
+            },
+          },
+        },
+      }));
+
+      jest.doMock("../utils/logger.js", () => ({
```

Comment:
Please move all mocks to a beforeEach, leaving inline will create duplications when more tests are added